src/JsonLd/Serializer/ItemNormalizer.php

```diff
@@ -306,6 +306,12 @@ private function denormalizeRelation(string $resourceClass, string $attributeNam
             return $this->serializer->denormalize($value, $className, self::FORMAT, $this->createRelationContext($className, $context));
         }
 
+        if (!is_array($value)) {
+            throw new InvalidArgumentException(sprintf(
+                'Expected IRI or nested object, %s given.',
+                $attributeName
```

Comment:
To know what property is wrong ?

If we give them juste the type it may be complicated to know where it come from no ?